pulp_container/app/registry_api.py

```diff
@@ -818,6 +818,29 @@ def put(self, request, path, pk=None):
         ):
             raise BlobInvalid(digest=config_blob.digest)
 
+        # both docker/oci format should contain layers, digest, media_type, size
+        layers = content_data.get("layers")
+        blobs = set()
+        for layer in layers:
+            media_type = layer.get("mediaType")
+            urls = layer.get("urls")
+            digest = layer.get("digest")
+            if (
+                media_type
+                in (
+                    models.MEDIA_TYPE.FOREIGN_BLOB,
+                    models.MEDIA_TYPE.FOREIGN_BLOB_OCI,
+                )
+                and not urls
+            ):
+                raise ManifestInvalid(digest=manifest_digest)
+            if media_type not in (
+                models.MEDIA_TYPE.REGULAR_BLOB,
+                models.MEDIA_TYPE.REGULAR_BLOB_OCI,
+            ):
+                raise BlobInvalid(digest=digest)
```

Comment:
I was looking here https://docs.docker.com/registry/spec/api/#errors-2

specifically at:

```


MANIFEST_INVALID | manifest invalid | During upload, manifests undergo several checks ensuring  validity. If those checks fail, this error may be returned, unless a  more specific error is included. The detail will contain information the  failed validation.
-- | -- | --


```

They say one can return a more specific error `If those checks fail, this error may be returned, unless a  more specific error is included`
Otherwise we'd just return 'manifest invalid digest=xxxx' error? you have any preference?

Reply:
I see inconsistency in the raise errors though, I need to fix that

Reply:
Alright so based on the docs, it is ok to raise blobinvalid during put of the manifest because it will create a 
400 response with the blob_unknown message.
It is also ok to raise manifestinvalid and it will create a 400 response with manifest_invalid message.
It is not ok to raise blobnotfound during put of manifest because that on creates a 404 response for blob.